x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverStep.java

```diff
@@ -30,6 +30,11 @@ public RolloverStep(StepKey key, StepKey nextStepKey, Client client) {
         super(key, nextStepKey, client);
     }
 
+    @Override
+    public boolean isRetryable() {
```

Comment:
I think we should do this in a separate PR. We still need to solve the complexity around retrying `AsyncAction` steps, which will require slightly different execution than the other step types, due to their exactly-once re-invocation.

Reply:
I think we satisfy the exactly-once re-invocation already. A failed `AsyncActionStep` will be moved into the `ErrorStep` on failure and on the next ILM periodic loop we'll move the index policy back into the failed step (the `AsyncActionStep`) which will be executed on the next async steps execution cycle (on a master change event or the subsequent periodic ILM loop). Am I missing something here? I'm happy to separate the PRs either way

Reply:
`AsyncActionStep`s don't get executed by the periodic loop invocation, meaning they will never be executed except on a master change event (which should be exceedingly rare). I was saying that we need to figure out the best way to make sure we can execute `AsyncAction` steps when they do an automatic retry.

To figure out this logic I was thinking a different PR would be better since it would likely mean extra testing for however we decide we want invoke `AsyncAction` steps (it may be as simple as adding the invocation in the `onProcessed` method of the `ilm-retry-failed-step` cluster state update, but we'll have to test).

Reply:
Ah, you're right Lee, I mixed the `AsyncWaitStep` with the `AsyncActionStep` (async and step was all my brain read)